adadamp/_dist.py

```diff
@@ -251,9 +251,16 @@ def score(self, X, y):
         with torch.no_grad():
             _loss = 0
             for Xi, yi in loader:
+
                 Xi, yi = Xi.to(self.device), yi.to(self.device)
                 y_hat = self.module_.forward(Xi)
-                _loss += self.loss_(yi.reshape(-1, 1), y_hat).item()
+                
+                print("Xi shape:", Xi.shape)
+                print("yi shape:", yi.shape)
+                print("yi re-shape:", yi.reshape(-1, 1).shape)
+                print("y-hat:", y_hat.shape) # Different 
```

Comment:
I am having a bug here. My targets I pass in via the `y` variable have 1 element per item, ie the expected class as an integer.
However, the model returns 10 element per item, which I expect is a prediction for each class. How do I convert y-hat so it is the same shape? Is it just the largest element in each output item?

Reply:
The docs for NLLLoss (https://pytorch.org/docs/stable/generated/torch.nn.NLLLoss.html) say that the first element should be two dimensional and the second input should be one dimensional. The example they give looks like this:

``` python
loss = nn.NLLLoss()
input = torch.randn(3, 5, requires_grad=True)
target = torch.tensor([1, 0, 4])
output = loss(model(input), target) 
```

I think `yi` and `y_hat` below should be swapped.